src/integrationTest/java/uk/gov/hmcts/reform/rse/idam/simulator/controllers/IdamSimulatorControllersHappyPathTest.java

```diff
@@ -194,6 +194,33 @@ public void returnOpenIdToken() throws Exception {
             .updateTokenInUser(A_USER_NAME, TOKEN);
     }
 
+    @DisplayName("Should return an open id token from code")
+    @Test
+    public void returnOpenIdTokenFromCode() throws Exception {
+        when(simulatorService.generateAuthTokenFromCode(anyString(), anyString(), anyString())).thenReturn(TOKEN);
```

Comment:
Can we use an actual simulator service instance instead of a mock? I appreciate the other tests mock it out but I think they should be fixed to use a real instance too.

Reply:
Yeah agreed - I've had to create a new test class for it though and move my tests into there.

Worth it as I actually found a pre-existing bug when I wrote the negative test. Hence the change to generateAuthTokenFromCode.